core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -586,24 +632,36 @@ object ConsumerGroupCommand extends Logging {
 
     def deleteGroups(): Map[String, Errors] = {
       val groupsToDelete = opts.options.valuesOf(opts.groupOpt).asScala.toList
-      val result = adminClient.deleteConsumerGroups(groupsToDelete)
-      val successfullyDeleted = result.filter {
+      val deletedGroups = adminClient.deleteConsumerGroups(
+        groupsToDelete.asJava,
+        withTimeoutMs(new DeleteConsumerGroupsOptions)
+      ).deletedGroups().asScala
+
+      val result = deletedGroups.mapValues { f =>
+        Try(f.get) match {
+          case _: Success[_] => Errors.NONE
+          case Failure(ee: ExecutionException) => Errors.forException(ee.getCause)
```

Comment:
I'm not sure it's reasonable to assume we will have an `ApiException`. My concern is that we will lose information i the exception gets mapped to UNKNOWN_ERROR. I'm wondering if we could skip the translation to error code and use the exception directly.

Reply:
``deleteGroups()`` returns a ``Map[String, Errors]`` -  https://github.com/apache/kafka/blob/b68766017f11afd7e541fa84378d5e616cf1e81a/core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala#L587

I can change it to return ``Map[String, Throwable]`` and update all related tests (such as ``DeleteConsumerGroupsTest`` that expects certain errors to happen e.g. ``Errors.GROUP_ID_NOT_FOUND``). If I do this, output for a delete group failure would change a bit. For example: 

```
Error: Deletion of some consumer groups failed:
* Group 'test.group' could not be deleted due to: java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.GroupNotEmptyException: The group is not empty
```

I will update the PR soon.